server/src/main/java/org/apache/druid/server/coordinator/CoordinatorDynamicConfig.java

```diff
@@ -100,8 +101,12 @@ public CoordinatorDynamicConfig(
     this.killDataSourceWhitelist = parseJsonStringOrArray(killDataSourceWhitelist);
     this.killPendingSegmentsSkipList = parseJsonStringOrArray(killPendingSegmentsSkipList);
     this.maxSegmentsInNodeLoadingQueue = maxSegmentsInNodeLoadingQueue;
-    this.maintenanceList = parseJsonStringOrArray(maintenanceList);
-    this.maintenanceModeSegmentsPriority = Math.min(10, Math.max(0, maintenanceModeSegmentsPriority));
+    this.historicalNodesInMaintenance = parseJsonStringOrArray(historicalNodesInMaintenance);
+    Preconditions.checkArgument(
+        nodesInMaintenancePriority >= 0 && nodesInMaintenancePriority <= 10,
```

Comment:
Does the priority of 0 make sense?

Reply:
It allows to suppress maintenance for a time, without clearing the maintenance list.